docs/language-server/protocol-project-manager.md

```diff
@@ -304,22 +381,377 @@ interface ProjectListSampleResponse {
 
 TBC
 
+## Action Progress Reporting
+
+### `task/started`
+
+Indicates that a long running task has been started.
+
+Currently only used when missing components are being installed to show
+installation progress.
+
+- **Type:** Notification
+- **Direction:** Server -> Client
+- **Connection:** Protocol
+- **Visibility:** Public
+
+#### Parameters
+
+```typescript
+interface TaskStartNotification {
+  taskId: UUID;
+  taskName: String;
+}
+```
+
+### `task/progress-update`
+
+Indicates a progress update for a specific task.
+
+- **Type:** Notification
+- **Direction:** Server -> Client
+- **Connection:** Protocol
+- **Visibility:** Public
+
+#### Parameters
+
+```typescript
+interface TaskUpdateNotification {
+  taskId: UUID;
+
+  /** Optional message explaining current status of the task. */
+  message?: String;
+
+  /** Indicates amount of progress, for example: count of processed bytes. */
+  done: Int;
+
+  /** Indicates total expected progress.
+   *
+   * May be missing, as it is not always known, for example when downloading a
+   * file of unknown size or waiting on a lock.
+   */
+  total?: Int;
```

Comment:
I think this should go to `task/started`. it should never change during progress, and we will immediately know if we have to display normal, or indeterminate progress bar.

Reply:
It shouldn't change indeed, but the issue is that the total progress may not be available immediately when the task is started - for example when downloading a file we may not know its size before we actually start the request and receive the headers.

But as that is an implementation detail, so I see no harm in simply postponing the `task/started` message until the total is known (i.e. in the current implementation, until the first progress update).

Reply:
> But as that is an implementation detail, so I see no harm in simply postponing the task/started message until the total is known (i.e. in the current implementation, until the first progress update).

Yes, actually we can't do much with single "task/start" message without any hint what progress bar we ought to display.